cmd/nginx-ingress/main.go

```diff
@@ -248,6 +251,17 @@ func main() {
 		glog.Fatal("NGINX App Protect support is for NGINX Plus only")
 	}
 
+	if *appProtectLogLevel != appProtectLogLevelDefault && !*appProtect && !*nginxPlus {
```

Comment:
I know that DOS is implemented the same way, but I find it a bit redundant and I think it also increases the complexity of the code

I feel something like this would be better
```
if *appProtect {
	if *appProtectLogLevel != appProtectLogLevelDefault {
		...
	}
	// all the other future checks
}
```
but maybe we can refactor both later. Do you think it adds much value to check those other flags if AP it's not enabled, instead of just ignoring it?

Anyway at this point in the code you also already checked that it's nginxPlus so there's no need to check it again

Reply:
Yeah, frankly speaking there is not much of a value in checking all flags if AP is not enabled. But It also sounds reasonable to me to refactor later in order to have the same approach for AP and DOS.